ThreadPool.hpp

```diff
@@ -0,0 +1,86 @@
+#ifndef THREAD_POOL_H
```

Comment:
Renaming the file is completely unnecessary move. And if you have made any changes to the header file beside the renaming, they are now invisible to most people.

Reply:
OK, It is better not move.